src/components/login/login.js

```diff
@@ -67,9 +67,58 @@ class LoginComponent {
   }
 
   onError(state, err) {
-    this.error = err;
-    this.state = state;
-    this.props.isSubmitting = false;
+    switch (err.action) {
+      case 'factor_challenge':
+        this.viewManager.showChallengeMfa({
+          section: 'challenge',
+          state: err.state,
+          selectedFactor: {
+            id: err.factor.type.toLowerCase(),
+            ...err.factor
+          },
+          onComplete: () => {
+            if (this.autoClose) {
+              this.autoClose();
+            }
+          }
+        });
+        break;
+
+      case 'factor_enroll':
+        this.viewManager.showEnrollMfa({
+          section: 'enroll',
+          state: err.state,
+          factors: err.allowedFactorTypes.map((id) => {
+            return {
+              id: id
```

Comment:
This is because the factors array that we provide expects an array of objects (with id identifying the factor). And what we're receiving here just an array of strings. So to reuse the same functionality, I'm remapping them to match the same structure.